astrocut/cube_cut.py

```diff
@@ -36,11 +37,23 @@ class CutoutFactory():
     Future versions will include more generalized cutout functionality.
     """
 
-    def __init__(self):
+    def __init__(self, threads: Union[int, None] = 1):
         """
         Initialization function.
+
+        Parameters
+        ----------
+        table_data : `~astropy.io.fits.fitsrec.FITS_rec`
+            The cube image header data table.
+
+        threads : int or `None`, default=1
+            Number of threads to use when making cutouts.
+            <=1 disables the threadpool, > 1 sets to the specified number of threads,
+            None uses `ThreadPoolExecutor`'s default: cpu_count + 4, limited to max of 32
```

Comment:
>  it sounds like this branch is using concurrent version 3.8

do you mean 3.8 and above?  or do you mean 3.8 specifically?

I don't think there's anything specific about 3.8 here.

> I find it a bit confusing that an input of None and an input of <=1 (for example 0) would not configure the number of threads the same way. 

Since the interface to `concurrent.futures` sets max_threads to be automatic when `max_workers=None`, I was sort of thinking it made sense to make our parameter match that. But I could change the auto behavior to only work for the string "auto" instead if that makes more sense. If I switch to "auto" then I guess `None` could instead be no multithreading? Or would you prefer I remove None from the type annotation entirely?

Reply:
we only support 3.8 and above so I think this should be fine. the param uses "auto" now.